### PR TITLE
Crypto by Grey Martin

### PR DESCRIPTION
**Pull request by jtallieu**

### CODE REVIEW

File: chomps/handlers/crypto.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+import requests
+import simplejson
+from pprint import pprint
+from lib import ChompsHandler
+from datetime import datetime, timedelta
+
+from muzzle import muzzles
+
+symbols = {"BTC" : "bitcoin",
+"ETH" : "ethereum",
```
Comment (jtallieu) [inline chomps/handlers/crypto.py:13]: *Style*

See PEP08 for conventions on style.  This requires an indent (like I said, it's only style and does not affect anything)

```
x = {
    'item': 3,
    'name': 'Jimmy'
}
```

File: chomps/handlers/crypto.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+"CNX" : "cryptonex",
+"PART" : "particl",
+"QSP" : "quantstamp",
+"PAY" : "tenx",
+"EMC" : "emercoin",
+"LINK" : "chainlink",
+"CVC" : "civic",
+"GNO" : "gnosis-gno"}
+
+odd_names = {
```
Comment (greymartinb) [inline chomps/handlers/crypto.py:113]: well I never used the odd_names dict for a third check

File: chomps/handlers/crypto.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        channel = msg["channel"]
+        muzzles.setdefault(channel, default)
+        pprint("muzzled")
+        pprint(muzzles)
+        pprint(muzzles[channel] > datetime.now())
+        if muzzles[channel] > datetime.now():
+            return "still quiet time"
+        else:
+            pprint(match.groups()[0])
+            url= "https://api.coinmarketcap.com/v1/ticker/" + match.groups()[0]
```
Comment (jtallieu) [inline chomps/handlers/crypto.py:170]: Usually it's proper to use either `"https://api.coinmarketcap.com/v1/ticker/{}".format(match.groups()[0])` or the `%` method to format strings.

File: chomps/handlers/crypto.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+                    return ("I got nothing. coding is hard")
+            # data[0]["percent_change_1h"] = "5.6"         
+            holder = trend(data[0]["percent_change_1h"])
+            holder_two = trend(data[0]["percent_change_24h"])
+            holder_three = trend(data[0]["percent_change_7d"])
+        return (">>>*Currency*: %s \n $ %s \n  %s *%s%%* _1hr_  %s *%s%%*  _1d_  %s *%s%%* _7d_ " %  ( data[0]["name"], data[0]["price_usd"],holder,data[0]["percent_change_1h"],holder_two,data[0]["percent_change_24h"],holder_three,data[0]["percent_change_7d"]) ) 
+        
+
+
+if __name__ == "__main__":
```
Comment (greymartinb) [inline chomps/handlers/crypto.py:202]: never changed the tests, wasn’t sure how they functioned

File: chomps/handlers/crypto.py [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+                    pprint("=========")
+                else:
+                    pprint(url)
+                    pprint(resp.status_code) 
+                    return ("I got nothing. coding is hard")
+            # data[0]["percent_change_1h"] = "5.6"         
+            holder = trend(data[0]["percent_change_1h"])
+            holder_two = trend(data[0]["percent_change_24h"])
+            holder_three = trend(data[0]["percent_change_7d"])
+        return (">>>*Currency*: %s \n $ %s \n  %s *%s%%* _1hr_  %s *%s%%*  _1d_  %s *%s%%* _7d_ " %  ( data[0]["name"], data[0]["price_usd"],holder,data[0]["percent_change_1h"],holder_two,data[0]["percent_change_24h"],holder_three,data[0]["percent_change_7d"]) )
```
Comment (greymartinb) [inline chomps/handlers/crypto.py:198]: need to change that last return on 198 to be across multiple lines for legibility

File: chomps/handlers/crypto.py [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+# pic_one = ":chart_with_upwards_trend:"
+# pic_two = ":chart_with_upwards_trend:"
+# pic_three = ":chart_with_upwards_trend:"
+
+pprint(muzzles)
+
+default =datetime(2018, 1, 16, 4, 15, 15, 362441)
+
+
+def trend(percent):
```
Comment (jtallieu) [inline chomps/handlers/crypto.py:147]: Nice use of function to get the icon.

File: chomps/handlers/crypto.py (hunk 5 above)
Comment (jtallieu) [inline chomps/handlers/crypto.py:198]: yes.  Split lines please.

**Comment by jtallieu** [comment]
Nice.  Only room for improvement would be to load all the quotes instead of trying to map the ID's.  Each time you get a request for a quote, you consult a local cached copy of the dict and search for a match on either (id, symbol, name).  If the data was loaded 2 minutes ago, get a fresh copy from the API then search the fresh results.  Eliminates a maintenance point when another coin cracks the top 100.

Basically _index_ (aka make a dict of id:result) the results of the API call and use that to get your data.  You could even say if getting a fresh copy failed, you could at least give the last results you got and astrik it with (delayed by x hours)
